timezones/test_timezones.py

```diff
@@ -4,67 +4,63 @@
 import datetime
 from timezones import zones, tz_utils, tz_rendering
 
-GEOIP_DATA_LOCATION = '/usr/local/geo_ip/GeoIP2-City.mmdb'
+GEOIP_DATA_LOCATION = "/usr/local/geo_ip/GeoIP2-City.mmdb"
 
 
-def test_sort():
-    # +04:00 is bigger than +04:30
-    assert zones._sort_by_tzoffset('+0400', '+0430') == -1
-
-    # +04:00 equal to +04:00
-    assert zones._sort_by_tzoffset('+0400', '+0400') == 0
+def assert_is_lower(offset_a, offset_b):
+    # Force a real sort to happen
```

Comment:
It's nice to have this comment here, but it's not obvious why the code below forces a "real sort". Can you please explain it shortly here?

Reply:
Sure. 

While writing the `_tz_offset_key` function, I realized there would be a problem for `"-0030"` and `"+0030"` if I only did `hours = -hours` for the negative case (as both would evaluate to a sorting key of `(0, 30)`), but I wasn't sure how Python would handle that.

Turns out that it's pretty smart, and very lazy: if the list appears ordered, nothing happens. So, `sorted(["-0030", "+0030"]) == ["-0030", "+0030"]`, and `sorted(["+0030", "-0030"]) == ["+0030", "-0030"]`.

By creating the list out of order, the sort algorithm really has to run and do something: it can't just give up and say "we're good!". This a *slightly* more realistic use case than just having an already sorted list of 2 elements.

And while writing this I realized that if the list is in reverse order, another optimization of Python's Timsort algorithm will kick in and just reverse it, so I'll edit that again 😅 I'll add more comments in the code this time.

Reply:
> add more comments in the code this time.

That was my point when asked for an explanation. The _here_ was actually referring to the code.

Thanks for the whole explanation here, tho. It's quite interesting, and I was not aware of that corner case!